docs/docs/guides/operate/managing-concurrency.md

```diff
@@ -107,18 +101,416 @@ concurrency:
         limit: 10
 ```
 
+## Job-Level Concurrency Configuration
+
+### Configuring Executors for Concurrency Control
+
+You can control concurrency at the job level by configuring executors. This is essential for controlling how many ops run simultaneously within a single job execution.
+
+#### Using the Multiprocess Executor
+
+The multiprocess executor allows you to control concurrency within a job run:
+
+```python
+from dagster import define_asset_job, multiprocess_executor
+
+# Configure the multiprocess executor with concurrency limits
+configured_multiprocess_executor = multiprocess_executor.configured(
+    {
+        "max_concurrent": 3,  # Maximum 3 concurrent processes
+        "tag_concurrency_limits": [
+            {
+                "key": "database",
+                "value": "postgres",
+                "limit": 1,
+            },
+            {
+                "key": "compute_type", 
+                "value": "heavy",
+                "limit": 2,
+            }
+        ],
+    }
+)
+
+# Apply executor to job
+my_job = define_asset_job(
+    name="my_concurrent_job",
+    selection=["asset_1", "asset_2", "asset_3"],
+    executor_def=configured_multiprocess_executor,
+)
+```
+
+#### Alternative: Using Job Configuration
+
+You can also configure concurrency through job configuration:
+
+```python
+my_job = define_asset_job(
+    name="my_job",
+    selection=["asset_1", "asset_2", "asset_3"],
+    config={
+        "execution": {
+            "config": {
+                "multiprocess": {
+                    "max_concurrent": 2,
+                    "tag_concurrency_limits": [
+                        {
+                            "key": "resource_type",
+                            "value": "database",
+                            "limit": 1,
+                        }
+                    ],
+                }
+            }
+        }
+    },
+)
+```
+
+### Asset-Level Concurrency with Tags
+
+Control concurrency for specific assets using tags:
+
+```python
```

Comment:
This example should also be put put in a file in the `docs_snippets/docs_snippets/guides/operate/managing_concurrency` folder and referenced with `<CodeExample>`